### ISSUE TITLE
Example README usage does not work?

### ISSUE THREAD

**Issue by alhassy** [tecosaur/org-knit-action#3]
Thanks for making this!

When I actually try to use it there is no HTML colouring and, worse, it doesn't actually push to the `gh-pages` branch. Perhaps there is setup missing from org-knit's README?

Here's some of what I see, ([full log here](https://github.com/alhassy/del/runs/4126931191?check_suite_focus=true)),

```
Starting org-knit
configuration
export and tangle
  1 file to process: index.org
   ✓ index.org tangled sucessfully
     Contacting host: fniessen.github.io:443
     Tangled 1 code block from index.org
   ✓ index.org exported to html sucessfully
     Contacting host: fniessen.github.io:443
     Cannot fontify source block (htmlize.el >= 1.34 required)
     Cannot fontify source block (htmlize.el >= 1.34 required)
   ✓ index.org exported to pdf sucessfully
     Contacting host: fniessen.github.io:443
     Processing LaTeX file index.tex...
     PDF file produced.
pushing
  Configure repository to push to gh-pages
  error: pathspec 'gh-pages' did not match any file(s) known to git
  Add remote, and stage files
  [main 4947973] Knit: 26c9a41e9fdd254b756b0922611a2cc909d936fb
   4 files changed, 350 insertions(+)
   create mode 100644 index.html
   create mode 100644 index.pdf
   create mode 100644 index.tex
   create mode 100644 woah.el
  Commited: "Knit: 26c9a41e9fdd254b756b0922611a2cc909d936fb"
  error: src refspec gh-pages does not match any
  error: failed to push some refs to '***github.com/alhassy/del.git'
  Pushed.
Completed
```

**Comment by tecosaur** [comment]
Thanks for opening the issue. Looking at your log, it seems like the bug comes from this bit of the code
https://github.com/tecosaur/org-knit-action/blob/30d3e53ae0a1ff33985fc2fa3242fc65866b3018/knit/push.py#L27-L37

I'd be curious to know if this works with `force_orphan: yes`, do you think you could try that for me?

**Comment by sorpaas** [comment]
I'm facing the same issue and unfortunately force_orphan does not work either.

**Comment by jethrokuan** [comment]
I'm also facing this issue, and have sort of reached a conclusion on why this happens.

The TL;DR is that you can't do a `git branch` when there is 0 commit history:

```
~/projects/test
❯ git init
Initialized empty Git repository in /Users/jethro/projects/test/.git/

test on  main
❯ git branch gh-pages
fatal: Not a valid object name: 'main'.

test on  main
❯ git checkout gh-pages
error: pathspec 'gh-pages' did not match any file(s) known to git

test on  main
❯
```

This corresponds to the above `pathspec` error. Then, when we commit it is on the original main/master branch, and try to push to the `gh-pages` branch, we find that locally we don't have the `gh-pages` branch either.

One solution is to then use `git checkout --orphan gh-pages` to always ensure we get the correct branch for content. Can refer to https://github.com/tsunematsu21/actions-publish-gh-pages/blob/master/src/index.ts for a popular implementation.

I would make a PR, but it seems that workflow depends on a Docker image containing the python files, so that needs to be rebuilt too.

**Comment by tecosaur** [comment]
Thanks for tracking this down @jethrokuan. Feel free to make a PR, there's actually a GH action which *should* 🤞 rebuild and publish the docker image (if it fails I'll just do so manually).

https://github.com/tecosaur/org-knit-action/blob/master/.github/workflows/build_image.yml
